File: s10_Service/Service/app/src/main/java/com/omeryildizce/service/MainActivity.java
package com.omeryildizce.service;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    Button startService, stopService;
    EditText editText;
    Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initalize();
        intent = new Intent(getApplicationContext(), ServiceClass.class);
        service();
    }

    private void service() {
        startService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(intent);
            }
        });

        stopService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                stopService(intent);
            }
        });
    }

    private void initalize() {
        startService = findViewById(R.id.buttonStart);
        stopService = findViewById(R.id.buttonStop);
        editText = findViewById(R.id.editTextText);
    }
}

File: s08_ActivityCycle/ActivityCycle/app/src/main/java/com/omeryildizce/activitycycle/MainActivity.kt
package com.omeryildizce.activitycycle

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    lateinit var editText: EditText;
    lateinit var textView: TextView;
    private val TAG = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "onCreate: ")
        editText = findViewById(R.id.editTextText)
        textView = findViewById(R.id.textView)

    }


    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
        super.onSaveInstanceState(outState, outPersistentState)
        Log.d(TAG, "onSaveInstanceState: ")
        outState.putString("saved", textView.text.toString())
    }

    override fun onRestoreInstanceState(
        savedInstanceState: Bundle?,
        persistentState: PersistableBundle?
    ) {
        super.onRestoreInstanceState(savedInstanceState, persistentState)
        Log.d(TAG, "onRestoreInstanceState: ")
         var text=   savedInstanceState?.getString("save") ?: "Hello"

        textView.setText(text)

    }
    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart: ")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume: ")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause: ")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop: ")
    }

    override fun onDestroy() {
        Log.d(TAG, "onDestroy: ")
        super.onDestroy()
    }
    fun getName(view: View) {

        var nameInput: String = editText.text.toString()
        textView.setText(nameInput)

    }

    fun secondActivity(view: View) {
        var intent = Intent(this, MainActivity2::class.java)
        startActivity(intent)
        finish()
    }


}